soroban-env-common/tests/no_std/Cargo.toml

```diff
@@ -0,0 +1,13 @@
+[package]
+name = "test_no_std"
```

Comment:
I'm going to merge this to keep the release 🚂 going, but @graydon if you have any alternative ideas for how to test this here in this repo or general feedback with what I'm doing here, I'm all ears.

Note that I already tried making a integration test module that has the harness turned off that is nostd. That didn't work because tests have to be binaries and this needs to be a lib.